cms/combine/cms_data.c

```diff
@@ -0,0 +1,1063 @@
+	free(data->pos);
+	free(data->freq1);
+	free(data->ihh0);
+	free(data->ihh1);
+	free(data->ihs_unnormed);
+	free(data->ihs_normed);
+	free(data->lastcol);	
+	data->nsnps = 0;
+} //end method
+
+/*************************/
+/***SCORE DISTRIBUTIONS***/
+/*************************/
+
+void get_likes_data(likes_data* data, char filename[]){
+	const int line_size = 15000000; 
+	FILE *inf=NULL;
+	char *newLine, *token, *running;
+	int ibin, itoken;
+
+	newLine = malloc((line_size+1) * sizeof(char));
+	assert(newLine != NULL); 
+
+	data->nbins = 0;
+	data->start_bin = NULL; 
+	data->end_bin = NULL;
+	data->probs = NULL;
+	fprintf(stderr, "\tloading likes tables from ");
+	fprintf(stderr, filename);
+	fprintf(stderr, "\n");
+
+	inf = fopen(filename, "r");
+	assert(inf != NULL);
+	fgets(newLine, line_size, inf); //strip
+	while (fgets(newLine, line_size, inf) != NULL){
+		data->nbins++;
+	}
+
+	data->start_bin = malloc(data->nbins * sizeof(double));
+	data->end_bin = malloc(data->nbins * sizeof(double));
+	data->probs = malloc(data->nbins * sizeof(double));
+	fclose(inf);
+
+	inf = fopen(filename, "r");
+	assert(inf != NULL);
+	ibin = 0;
+	while (fgets(newLine, line_size, inf) != NULL){
+		for (running = newLine, itoken = 0; (token = strsep(&running, "\t")) != NULL; itoken++) {
+			//fprintf(stderr, token);
+		 // fprintf(stderr, "\n%d\n", itoken);
+			if (itoken == 0) {
+				data->start_bin[ibin] = atof(token);
+			}
+			else if (itoken ==1){
+				 data->end_bin[ibin] = atof(token);
+			}
+			else if (itoken ==2){
+				 data->probs[ibin] = atof(token);
+				 break;
+			} 
+		} // end for loop
+		ibin++;
+	} // end while loop
+
+	fclose(inf);
+	free(newLine);
+} //end method
+
+void free_likes_data(likes_data* data) {
+	if (data == NULL) {return;}
+	data->nbins = 0;
+	free(data->start_bin);
+	free(data->end_bin);
+	free(data->probs);
+} //end method
+
+/**********************/
+/***POP COMPARISONS****/
+/**********************/
+
+void get_popComp_data_multiple_region(popComp_data_multiple* data, int argc, char *argv[]){
+  ///collates snps for an arbitrary number of populations compared to a putative selpop,
+  ///and returns component scores
+	const int line_size = 15000000; 
+	FILE *inf=NULL;
+	char *newLine, *token, *running;
+	char infilebase[256], infilesuffix[256], infilename[512];
+	int isnp, jsnp, itoken, iComp, nComparisons, totNsnp, thisPhysPos, nunique; //maxNsnp, i, 
+	int startBp, endBp;
+	popComp_data data_sing;
+	int *allSnps, *allUniqueSnps;
+	int iLine;
+
+	//////////////////
+	/// INITIALIZE ///
+	//////////////////
+
+	newLine = malloc((line_size+1) * sizeof(char));
+	assert(newLine != NULL); 
+
+	data->nsnps = 0;
+	data->physpos = NULL;
+	data->genpos = NULL;
+	data->daf_selpop = NULL;
+	data->delDAF = NULL;
+	data->fst = NULL;
+	data->xp_normed = NULL;
+	data->ihs_normed = NULL;
+	data->delihh_normed = NULL;
+
+	startBp = atoi(argv[1]);
+	endBp = atoi(argv[2]);
+
+	sprintf(infilebase, "/idi/sabeti-scratch/jvitti/synth/cms_pairs/");
```

Comment:
Does it make sense to convert the hard-coded paths into parameterized ones? Or save that for the future? Ideally they would be passed in via something like [getopt](https://www.gnu.org/software/libc/manual/html_node/Getopt.html). @notestaff may have better guidance on the subject. I do like the selscan [solution](https://github.com/szpiech/selscan/blob/master/src/selscan-main.cpp), and it's licensed under GPLv3, so it could be reused in this repo (if licensed appropriately).